nilearn/signal.py

```diff
@@ -641,16 +661,46 @@ def _filter_signal(signals, confounds, filter, low_pass, high_pass, t_r):
             confounds = butterworth(confounds, sampling_rate=1. / t_r,
                                     low_pass=low_pass, high_pass=high_pass)
     elif filter == 'cosine':
-        from .glm.first_level.design_matrix import _cosine_drift
+        from nilearn.glm.first_level.design_matrix import _cosine_drift
         frame_times = np.arange(signals.shape[0]) * t_r
-        cosine_drift = _cosine_drift(high_pass, frame_times)
-        if confounds is None:
-            confounds = cosine_drift.copy()
-        else:
-            confounds = np.hstack((confounds, cosine_drift))
+        # remove constant, as the signal is mean centered
+        cosine_drift = _cosine_drift(high_pass, frame_times)[:, :-1]
+        confounds = _check_cosine_by_user(confounds, cosine_drift)
```

Comment:
Also, I might be missing something but I was expecting the signals to be modified somehow by this function since it is called `_filter_signal`. It is the case when `filter == butterworth`, but doesn't seem to be the case with cosine.
I just wanted to check that this is expected as it feels a little surprising when reading the code.